client/my-sites/site-settings/media-settings.jsx

```diff
@@ -17,111 +18,182 @@ import CompactFormToggle from 'components/forms/form-toggle/compact';
-					/>
-			</FormFieldset>
-			<FormFieldset className="media-settings__formfieldset has-divider is-top-only">
-				<div className="media-settings__info-link-container site-settings__info-link-container">
+		return isVideoPressAvailable
+			? <FormFieldset className="site-settings__formfieldset has-divider is-top-only">
+				<div className="site-settings__info-link-container">
 					<InfoPopover position="left">
-						<ExternalLink target="_blank" icon={ true } href="https://jetpack.com/support/carousel" >
-							{ translate( 'Learn more about Carousel.' ) }
+						<ExternalLink target="_blank" icon href="https://jetpack.com/support/videopress/" >
+							{ translate( 'Learn more about VideoPress.' ) }
 						</ExternalLink>
 					</InfoPopover>
 				</div>
 				<JetpackModuleToggle
 					siteId={ siteId }
-					moduleSlug="carousel"
-					label={ translate( 'Transform standard image galleries into full-screen slideshows' ) }
-					disabled={ isRequestingSettings || isSavingSettings }
-					/>
-				<div className="media-settings__module-settings site-settings__child-settings">
-					<CompactFormToggle
-						checked={ fields.carousel_display_exif || false }
-						disabled={ isRequestingSettings || isSavingSettings || ! carouselActive }
-						onChange={ handleAutosavingToggle( 'carousel_display_exif' ) } >
-						{ translate( 'Show photo metadata in carousel, when available' ) }
-					</CompactFormToggle>
-					<FormLabel className={ labelClassName } htmlFor="carousel_background_color">
-						{ translate( 'Background color' ) }
-					</FormLabel>
-					<FormSelect
-						name="carousel_background_color"
-						id="carousel_background_color"
-						value={ fields.carousel_background_color || 'black' }
-						onChange={ onChangeField( 'carousel_background_color' ) }
-						disabled={ isRequestingSettings || isSavingSettings || ! carouselActive } >
-						<option value="black" key="carousel_background_color_black">
-							{ translate( 'Black' ) }
-						</option>
-						<option value="white" key="carousel_background_color_white">
-							{ translate( 'White' ) }
-						</option>
-					</FormSelect>
-				</div>
+					moduleSlug="videopress"
+					label={ translate( 'Enable fast, ad-free video hosting' ) }
+					disabled={ isRequestingOrSaving }
+				/>
 			</FormFieldset>
-		</Card>
-	);
-};
+			: null;
+	}
 
-MediaSettings.propTypes = {
-	carouselActive: PropTypes.bool.isRequired,
-	isSavingSettings: PropTypes.bool,
-	isRequestingSettings: PropTypes.bool,
-	siteId: PropTypes.number.isRequired,
-	handleAutosavingToggle: PropTypes.func.isRequired,
-	onChangeField: PropTypes.func.isRequired,
-	fields: PropTypes.object
-};
+	renderVideoUpgradeNudge() {
+		const {
+			isVideoPressAvailable,
+			translate,
+		} = this.props;
+
+		return isVideoPressAvailable
+			? null
+			: <Banner
+				event={ 'jetpack_video_settings' }
+				feature={ FEATURE_VIDEO_UPLOADS_JETPACK_PREMIUM }
```

Comment:
Does it make sense to highlight the feature only in the premium plan, and not in the pro plan? cc @richardmuscat